rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/client/connection/AssistCreatingConnectionParams.java

```diff
@@ -27,11 +27,14 @@ public class AssistCreatingConnectionParams {
   private final String tokenName;
   @Nullable
   private final String tokenValue;
+  @Nullable
+  private final String organizationKey;
 
-  public AssistCreatingConnectionParams(String serverUrl, @Nullable String tokenName, @Nullable String tokenValue) {
+  public AssistCreatingConnectionParams(String serverUrl, @Nullable String tokenName, @Nullable String tokenValue, @Nullable String organizationKey) {
```

Comment:
For APIs that relate to SQ or SC, we usually need 2 mutually exclusive sets of info: 
* for SQ, the server URL
* for SC, the organization key

So far, in various APIs, we have used the `Either` type to expose exactly the data needed. The advantage is that it documents what should be present in each case, better than having 2 nullable fields in the same DTO. You can check, for example, `CheckSmartNotificationsSupportedParams`.

Reply:
Yes, it's a good practice. I'll apply it